src/entities/model.ts

```diff
@@ -14,7 +15,24 @@ export { ModelComponent, ModelTaskMetadata } from "./modelTaskMetadata";
  * A model observer: a function that takes a change set as input
  * and produces a list of tasks to process as output.
  */
-type ModelObserver = (changeBuffer: Immutable.Set<ModelComponent>) => Array<ModelTask>;
+export class ModelObserver {
```

Comment:
Quick suggestion to avoid breaking other people's forks: update `registerObserver` to take either a new-style `ModelObserver` or an old one.

```ts
public registerObserver(
    observer: ModelObserver
        | (changeBuffer: Immutable.Set<ModelComponent>) => Array<ModelTask>): void {

    if (observer instanceof ModelObserver) {
        this.observers.push(observer);
    } else {
        this.observers.push(new ModelObserver(observer));
    }
}
```